Send PagerDuty notifications if a test fails

nigeldeakin opened the pull request:
This PR extends these tests to send PagerDuty notifications if a test fails.

This has been done by adding an after-step at the end of every pipeline that calls the `nigeldeakin/pagerduty-notifier` step.

To make these changes work, the following environment variables need to be set:

* `PAGERDUTY_NOTIFY_ON` needs to be set to `failed` (so we only get notifications when pipelines fail)
* `PAGERDUTY_BRANCH` needs to be set to `master`(so we only get notifications for runs on the main branch)
* `PAGERDUTY_SERVICE_KEY` needs to be set to an appropriate PagerDuty service key. This will send notifications to `#wercker-alerts` and `#wercker-alerts-stage` as appropriate.

Notes:

* We don't want users to fork this repo and add `echo` statements to dump out `PAGERDUTY_SERVICE_KEY`. This means that we may defer setting `PAGERDUTY_SERVICE_KEY` until https://jira.oraclecorp.com/jira/browse/WRKR-886 or https://jira.oraclecorp.com/jira/browse/WRKR-885 are done.

* The after-steps in those pipelines that (1) use alpine and (2) use the docker internal steps will not be run until https://jira.oraclecorp.com/jira/browse/WRKR-1050 is fixed. 

* In addition the notification will contain an invalid run URL until https://jira.oraclecorp.com/jira/browse/WRKR-1048 is fixed. 

However this PR can itself be merged safely without waiting for these issues to be fixed.

hatchan commented on wercker.yml:
```diff
@@ -95,8 +95,12 @@ build:
+          go test ./... 
+  after-steps:
+    - nigeldeakin/pagerduty-notifier@1.1.0:
+        service_key: $PAGERDUTY_SERVICE_KEY
```
if we're going to move this step to the wercker org, then we probably want to use dashes instead of underscores (might even work already though).

nigeldeakin commented:
I've updated `pagerduty-notifier` to use hyphens rather than underscores in its property names, in accordance with the naming convention for steps.

I've changed the reference to `nigeldeakin/pagerduty-notifier` to omit the version number, so we use the latest.

I will now merge this. This will leave `wercker-tests` in the following state:
* because of https://jira.oraclecorp.com/jira/browse/WRKR-1050 the after-steps will not be run on several of the pipelines (those that call internal docker steps)
* for the pipelines where the new after-steps are actually run, the `pagerduty-notifier` step will fail (harmlessly) until `PAGERDUTY_SERVICE_KEY` is set.
